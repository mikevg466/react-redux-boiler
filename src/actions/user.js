import types from './types';

export const setName = name => ({
  type: types.SET_NAME,
  name,
});

export const setGreeting = greeting => ({
  type: types.SET_GREETING,
  greeting,
});
