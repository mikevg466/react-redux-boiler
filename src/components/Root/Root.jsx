import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import logo from '../../assets/images/logo.svg';
import './Root.css';

/**
 * Root React Component containing the presentational Root
 * @extends PureComponent
 */
class RootComponent extends PureComponent {
  /**
   * React Render
   * @return {JSX}
   */
  render() {
    const { greeting, name, actions } = this.props;
    return (
      <div className='Root'>
        <header className='Root-header'>
          <img alt='logo' className='Root-logo' src={logo} />
          <p>
            {`${greeting} `}
            <input
              onChange={e => actions.setName(e.target.value)}
              value={name}
            />
            {'!'}
          </p>
          <a
            className='Root-link'
            href='https://reactjs.org'
            rel='noopener noreferrer'
            target='_blank'
          >
            Learn React
          </a>
        </header>
      </div>
    );
  }
}

RootComponent.propTypes = {
  name: PropTypes.string,
  greeting: PropTypes.string,
  actions: PropTypes.object,
};

export default RootComponent;
