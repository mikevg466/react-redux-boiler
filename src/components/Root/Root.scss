@import '../../stylesheets/modules/classNames';
@import '../../stylesheets/modules/colors';

.#{$project-name} {
  .root {
    text-align: center;

    .logo {
      @keyframes Root-logo-spin {
        from {
          transform: rotate(0deg);
        }
        to {
          transform: rotate(360deg);
        }
      }

      animation: Root-logo-spin infinite 20s linear;
      height: 40vmin;

      .header {
        background-color: #282c34;
        min-height: 100vh;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        font-size: calc(10px + 2vmin);
        color: white;
      }

      .link {
        color: $link-color;
      }
    }
  }
}
