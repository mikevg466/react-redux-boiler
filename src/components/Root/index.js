export { default as RootComponent } from './Root.jsx';
