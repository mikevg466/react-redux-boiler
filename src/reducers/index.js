import { combineReducers } from 'redux';
import user from './user';

// Use ES6 object literal shorthand syntax to define the object shape
const rootReducer = combineReducers({
  user,
});

export default rootReducer;
