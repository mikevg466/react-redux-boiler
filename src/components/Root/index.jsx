import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import logo from '../../assets/images/logo.svg';
import './Root.scss';

/**
 * Root React Component containing the presentational Root
 * @extends PureComponent
 */
export default class RootComponent extends PureComponent {
  /**
   * React Render
   * @return {JSX}
   */
  render() {
    const { greeting, name, setName } = this.props;
    return (
      <div className='root'>
        <header className='header'>
          <img alt='logo' className='logo' src={logo} />
          <p>
            {`${greeting} `}
            <input onChange={e => setName(e.target.value)} value={name} />
            {'!'}
          </p>
          <a
            className='link'
            href='https://reactjs.org'
            rel='noopener noreferrer'
            target='_blank'
          >
            Learn React
          </a>
        </header>
      </div>
    );
  }
}

RootComponent.propTypes = {
  name: PropTypes.string,
  greeting: PropTypes.string,
  setName: PropTypes.func.isRequired,
};
