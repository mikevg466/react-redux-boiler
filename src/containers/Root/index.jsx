import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { setName } from '../../actions/user';
import Root from '../../components/Root';

/**
 * Root React Container to be initialized in the main index.js file
 * @extends Component
 */
export class RootContainer extends Component {
  /**
   * React Render
   * @return {JSX}
   */
  render() {
    const { name, greeting, setName } = this.props;
    return <Root greeting={greeting} name={name} setName={setName} />;
  }
}

RootContainer.propTypes = {
  name: PropTypes.string,
  greeting: PropTypes.string,
  setName: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  name: state.user.name,
  greeting: state.user.greeting,
});

export const mapDispatchToProps = {
  setName,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(RootContainer);
