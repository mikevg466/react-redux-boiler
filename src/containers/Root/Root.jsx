import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { UserActions } from '../../actions';
import { RootComponent } from '../../components/Root';

/**
 * Root React Container to be initialized in the main index.js file
 * @extends Component
 */
class RootContainer extends Component {
  /**
   * React Render
   * @return {JSX}
   */
  render() {
    return <RootComponent {...this.props} />;
  }
}

const mapStateToProps = ({ user }) => ({
  name: user.name,
  greeting: user.greeting,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(UserActions, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(RootContainer);
