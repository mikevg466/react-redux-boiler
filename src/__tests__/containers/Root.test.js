import React from 'react';
import { shallow } from 'enzyme';
import { RootContainer, mapStateToProps } from '../../containers/Root';
import { initialUserState } from '../../reducers/user';

const mockState = {
  greeting: initialUserState.greeting,
  name: initialUserState.name,
};
const mockDispatchers = {
  setName: jest.fn(),
};
const mockProps = {
  ...mockState,
  ...mockDispatchers,
};

describe('Root Container', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<RootContainer {...mockProps} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('outputs the expected state when passed a mock store', () => {
      const mockStore = {
        user: initialUserState,
      };
      const expectedState = {
        greeting: initialUserState.greeting,
        name: initialUserState.name,
      };
      expect(mapStateToProps(mockStore)).toEqual(expectedState);
    });
  });
});
