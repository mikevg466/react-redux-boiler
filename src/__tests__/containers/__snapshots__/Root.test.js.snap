// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Container renders correctly 1`] = `"<RootComponent greeting=\\"hello\\" name=\\"world\\" setName={[Function: mockConstructor]} />"`;
